src/routes/github/subscription-deferred-install/github-subscription-deferred-install-router.ts

```diff
@@ -0,0 +1,73 @@
+import { NextFunction, Request, Response, Router } from "express";
+import { extractSubscriptionDeferredInstallPayload, SubscriptionDeferredInstallPayload } from "services/subscription-deferred-install-service";
+import { Installation } from "models/installation";
+import { GithubServerAppMiddleware } from "middleware/github-server-app-middleware";
+import { GithubAuthMiddleware } from "routes/github/github-oauth";
+import {
+	GithubSubscriptionDeferredInstallGet
+} from "routes/github/subscription-deferred-install/github-subscription-deferred-install-get";
+import { booleanFlag, BooleanFlags } from "config/feature-flags";
+import {
+	GithubSubscriptionDeferredInstallPost
+} from "routes/github/subscription-deferred-install/github-subscription-deferred-install-post";
+
+const GithubSubscriptionDeferredInstallRouter = Router({ mergeParams: true });
+
+const subRouter = Router({ mergeParams: true });
+
+const INVALID_PAYLOAD_ERR = "Invalid payload";
+
+GithubSubscriptionDeferredInstallRouter.use("/request/:requestId", subRouter);
+
+const validatePayloadAndPopulateJiraHost = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+	let parsedPayload: SubscriptionDeferredInstallPayload | undefined;
+	try {
+		parsedPayload = await extractSubscriptionDeferredInstallPayload(req);
+	} catch (err) {
+		req.log.warn({ err }, "Cannot deserialize");
+		res.status(400).json({ error: INVALID_PAYLOAD_ERR });
+		return;
+	}
+
+	const installation = await Installation.findByPk(parsedPayload.installationIdPk);
+	if (!installation) {
+		req.log.warn("No installation");
+		res.status(400).json({ error: INVALID_PAYLOAD_ERR });
+		return;
+	}
+
+	if (!await booleanFlag(BooleanFlags.ENABLE_SUBSCRIPTION_DEFERRED_INSTALL, installation.jiraHost)) {
+		res.status(401).json({ error: "Feature is disabled" });
+		return;
+	}
+
+	res.locals.jiraHost = installation.jiraHost;
+	res.locals.installation = installation;
+	// DO NOT PUT jiraHost to session! This router is invoked by jira non-admins, that would give them admin perms!
+
+	return next();
+};
+
+const validateGitHubConfig = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+	const parsedPayload = await extractSubscriptionDeferredInstallPayload(req);
+
+	if (parsedPayload.gitHubServerAppIdPk != res.locals.gitHubAppConfig.gitHubAppId) {
+		req.log.warn("Wrong appIdPk");
+		res.status(400).json({ error: INVALID_PAYLOAD_ERR });
+		return;
+	}
+
+	return next();
+};
+
+subRouter.use(validatePayloadAndPopulateJiraHost);
```

Comment:
/nit
**IMPORTANT**

I just hate how we setup the routing in these nested behaviour. So un-readable. I just would love to see following in ONE SINGLE FILE!!

```
const router = app.router();
const routerConfig = {
   "/github": {
     "/deferInstallUrl": [
        GitHubServerAppMiddleware,
        GitHubAuthMiddlware, 
        {
           "GET": gitHubDeferUrlGet
        }
      ],
      "/configuration: [GitHubServerAppMiddleware, GitHubAuthMiddlware, {
           "GET": gitHubConfigGet
           "POST": gitHubConfigPost
      }"]
  },
  "jira": {
  }
};
router.use(routerConfig);
```

Reply:
**AND**!!
Inside the so-called `gitHubConfigPost`, it is actually just a function, take params return params, and none of the params are http related. There shouldn't be any `res.locals.xxx`. It should be just plain js object params. 

The extractor should be somewhere else.

Reply:
so /nit, just me ranting.

Reply:
Wow, Gary rage 😜

Reply:
I think we should strive for a good test coverage, thus reducing chances that somehting is wrong with this logic. For each endpoint check everything: when it returns 401, when it doesn't, etc. At least this is what I'm doing now :shrug: